src/main/java/net/starlark/java/syntax/NodePrinter.java

```diff
@@ -482,4 +476,18 @@ private void printExpr(Expression expr) {
         }
     }
   }
+
+  private void printClause(Clause clause) {
+    if (clause instanceof Comprehension.For) {
+      Comprehension.For forClause = (Comprehension.For) clause;
+      buf.append("for ");
+      printExpr(forClause.getVars());
+      buf.append(" in ");
+      printExpr(forClause.getIterable());
+    } else {
+      Comprehension.If ifClause = (Comprehension.If) clause;
+      buf.append("if ");
+      printExpr(ifClause.getCondition());
+    }
+  }
```

Comment:
In this particular case I personally find the code more readable without the early return, which destroys the symmetry between the branches. Ideally, this would be a switch over a sealed class, but we don't have those.

Reply:
I agree, the branches have equal weight and so 'else' is appropriate.